developer/src/kmc-ldml/src/compiler/keys.ts

```diff
@@ -113,95 +187,111 @@ export class KeysCompiler extends SectionCompiler {
       );
     } else if (hardwareLayers.length === 1) {
       const theLayers = hardwareLayers[0];
-      const { form } = theLayers;
+      const { formId } = theLayers;
       for (let layer of theLayers.layer) {
-        this.compileHardwareLayerToKmap(sections, layer, sect, form);
+        this.compileHardwareLayerToKmap(sections, layer, sect, formId);
       }
     } // else: TODO-LDML do nothing if only touch layers
 
     return sect;
   }
 
-  public loadFlicks(sections: DependencySections, sect: Keys) {
-    for (let lkflicks of this.keyboard3.keys.flicks) {
+  static addUsedGestureKeys(layerKeyIds: string[], keyBag: Map<string, LDMLKeyboard.LKKey>, usedKeys: Set<string>) {
+    for (let keyId of layerKeyIds) {
+      const key = keyBag.get(keyId);
+      if (!key) continue;
```

Comment:
Ditto should this be a warning?

Reply:
here, no, because it's already handled at the top of validate().